### ISSUE TITLE
DotNet watch should not trigger itself

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/aspnetcore#36306]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/DotNet-watch-should-not-trigger-itself/1518401)._

---
[severity:I'm unable to use this version]
I create a command-line project (ie: TestCommandLineProject), and go to the command line and run ```dotnet watch run``` and the project builds and runs the program.cs file and output is good. It then says ```Waiting for a file to change before restarting dotnet...```. and then within a few seconds, it rebuilds the project and repeats indefinitely without me making any code changes or saves to any files in the project.

The below is repeated continuously until [CTRL]+[C] is pressed.

```
watch : Waiting for a file to change before restarting dotnet...
watch : Building...
  Determining projects to restore...
  All projects are up-to-date for restore.
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  RiskPolicyResponseParser -> C:\Users\paul\Source\Repos\TestCommandLineProject\bin\Debug\net6.0\TestCommandLineProject.dll
watch : Started
Hello World!
watch : Exited
watch : Waiting for a file to change before restarting dotnet...
watch : Building...
  Determining projects to restore...
  All projects are up-to-date for restore.
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  RiskPolicyResponseParser -> C:\Users\paul\Source\Repos\TestCommandLineProject\bin\Debug\net6.0\TestCommandLineProject.dll
watch : Started
Hello World!
watch : Exited
watch : Waiting for a file to change before restarting dotnet...
```

---
#### Feedback Bot on 9/1/2021, 01:32 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by pranavkm** [comment]
Thank you for filing this issue. In order for us to investigate this issue, please provide a minimal project (ideally a GitHub repo) that illustrates the problem.

**Comment by daemogar** [comment]
Literally create a new console app. There are only two files (shown below) when you do and I did not modify anything at all. In my case, it did create a solution folder because I was doing this through VS 2022, but I also just copied those two files below into a separate folder and used vscode to view them and from the command line, ran ```dotnet watch run```.

Additional Process Clarification: it will run as you expect it. You will not think anything is wrong. Open the program.cs file in your IDE of choice, and save it. It will rebuild and run the project again. Great! That is what we expect, but here is where things go south. The watch is triggered again, with the same results, and then triggered again, with the same results. It never stops, unless you close the console or [CTRL]+[C] out of it. Which defeats the purpose.  :)

ExampleProject.csproj
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <Nullable>enable</Nullable>
  </PropertyGroup>

</Project>
```

Program.cs
```cs
// See https://aka.ms/new-console-template for more information

Console.WriteLine ("Hello, World!");
```

**Comment by pranavkm** [comment]
Thanks @daemogar. This issue has been addressed in the rc2 builds of the SDK. You could try out a nightly build of the installer from https://github.com/dotnet/installer#installers-and-binaries to verify this.

**Comment by daemogar** [comment]
I installed the nightly, but I think I was to quick at doing this. The nightly I installed, ```dotnet-sdk-6.0.100-rc.2.21460.6-win-x64```, does not have the fix. I will try again later.

However, after installing that version, I was required to add using System; to the ```Program.cs``` file. Not sure if I should expect that behavior or not with this or the next nightly build.

After installing, and before running ```dotnet watch run```, I ran ```dotnet --version``` and the result was ```6.0.100-rc.2.21460.6```, so it seems that the correct version of dotnet was used.

Program.cs
```cs
// See https://aka.ms/new-console-template for more information

using System;

Console.WriteLine ("Hello, World!");
```
